src/vm/profilerdiagnosticprotocolhelper.cpp

```diff
@@ -103,6 +103,14 @@ void ProfilerDiagnosticProtocolHelper::AttachProfiler(DiagnosticsIpc::IpcMessage
         goto ErrExit;
     }
     
+    // The diagnostic server is single threaded, so we do not have to worry about racing
```

Comment:
I'd recommend making the ProtocolHelper be responsible only for marshalling serialized buffer -> arguments and return value -> serialized buffer. For any logic related to the operation itself I'd put it inside LoadProfilerForAttach.

Reply:
Yeah, I think that makes more sense